src/File.php

```diff
@@ -803,6 +803,20 @@ public function onBeforePublish()
         }
     }
 
+    public function renameFile($newName)
```

Comment:
If you don't ever write the file you'll break any future reference to this file. Ensuring it immediately writes would fix this.